h/static/scripts/streamer.js

```diff
@@ -155,6 +172,53 @@ function Streamer($rootScope, annotationMapper, groups, session, settings) {
     _connect();
   };
 
+  function applyPendingUpdates() {
+    var updates = Object.values(pendingUpdates).filter(function (ann) {
+      // Ignore updates to annotations that are not in the focused group
+      return ann.group === groups.focused().id;
+    });
+    var deletions = Object.keys(pendingDeletions).map(function (id) {
+      return {id: id};
+    });
+
+    annotationMapper.loadAnnotations(updates);
+    annotationMapper.unloadAnnotations(deletions);
+
+    pendingUpdates = {};
+    pendingDeletions = {};
+  }
+
+  function countPendingUpdates() {
+    return Object.keys(pendingUpdates).length;
```

Comment:
The first issue you describe could be a bit confusing, but I don't really think it's the end of the world for now.

The second issue describe is present today, only without the benefit of an explicit action the user may understand led to the removal of the annotation to which they were replying.